sway/xdg_activation_v1.c

```diff
@@ -44,7 +49,13 @@ void xdg_activation_v1_handle_new_token(struct wl_listener *listener, void *data
 
 	struct sway_workspace *ws = seat_get_focused_workspace(seat);
 	if (ws) {
-		launcher_ctx_create(token, &ws->node);
+		struct launcher_ctx *ctx = launcher_ctx_create(token, &ws->node);
+		if (ctx) {
```

Comment:
As above, but first move this into launcher_ctx create. There you can record whatever necessary state.

Reply:
The assignment was here because it's an xdg-activation policy decision and hiding it in the `launcher_ctx_create` didn't seem right.
Well, if we go with separate flags for seat/surface, it would be reasonable to set the flags in `launcher_ctx_create`, but do the check in `xdg_activation_v1_handle_request_activate`.

Reply:
> The assignment was here because it's an xdg-activation policy decision

`can_activate` is a policy decision, but something like `had_focused_surface` is a plain fact about the compositor state. That is what I'm suggesting.

Reply:
Ok, latest update does that.